stan/math/prim/scal/meta/and.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_AND_HPP
+#define STAN_MATH_PRIM_SCAL_META_AND_HPP
+
+#include <type_traits>
+
+namespace stan {
+namespace math {
```

Comment:
I put the helper metaprograms for conjuction and disjunction to the stan::math namespace, as the metaprograms are in all sorts of namespaces. See debate in #1122.

Reply:
Are these just duplicating the C++17 types`conjunction` and `disjunction`?  E.g., see

https://en.cppreference.com/w/cpp/types/conjunction

There's a sizable type traits library now:

https://en.cppreference.com/w/cpp/header/type_traits

Reply:
Yes, this is duplicating the conjunction and disjunction for sub C++17 and will be replaced directly with them once Stan Math switches to C++17. As discussed in the [issue thread](https://github.com/stan-dev/math/issues/1122#issuecomment-468016606), we have 2 options to do something like this in C++14, that are described here: https://stackoverflow.com/questions/29603364/type-trait-to-check-that-all-types-in-a-parameter-pack-are-copy-constructible/29603857#29603857

Reply:
Yes, this is duplicating the conjunction and disjunction for sub C++17 and will be replaced directly with them once Stan Math switches to C++17. As discussed in the issue thread, we have 2 options to do something like this in C++14, that are described here: https://stackoverflow.com/questions/29603364/type-trait-to-check-that-all-types-in-a-parameter-pack-are-copy-constructible/29603857#29603857